Empire/ProductionQueue.cpp

```diff
@@ -307,9 +309,11 @@ ProductionQueue::ProductionItem::ProductionItem() :
 {}
 
 ProductionQueue::ProductionItem::ProductionItem(BuildType build_type_) :
-    build_type(build_type_),
-    name("PROJECT_BT_STOCKPILE")
-{}
+    build_type(build_type_)
+{
+    if (build_type_ == BT_STOCKPILE)
+        name = "PROJECT_BT_STOCKPILE";
```

Comment:
Can this be / is this ever shown to the user as a stringtable lookup? If so, should wrap in `UserStringNoOp`

Reply:
Is it  a problem to nest a UserStringNoOp like this: UserString(UserStringNoOp("MSG_BLABLA"))?

Because ProductionWnd wraps the item names like this: UserString(elem.item.name);

Reply:
If a string literal is already written in a `UserString` call then it doesn't also need a `NoOp` wrapper.

Reply:
So this stays the way it was. The buildings also store their stringtable keys in the name as simple text. UserStringification happens on use site.

Reply:
If it is written in a `UserString` call, it doesn't need `UserStringNoOp`. If it is assigned to a variable that is later passed into `UserString`, then the string literal it should be wrapped in `UserStringNoOp`.

Reply:
Why should i wrap a string in a UserStringNop at all? For documentation of intent?

Hm. I think i dont quite grok this. As far as i understand the intention, UserStringNop should be used instead of UserString if you do not want any stringtable lookup to happen (e.g. to preserve a key). Implementation-wise it does get stripped away anyway - so doesnt have any effect at all?

As far as I can see, multiple nested UserString calls like UserString(UserString(key)) would usually do the wrong thing (using the value of key as key for another lookup). In this case as the UserString call happens on the use site of ProductionItem, I should return the key. 

I think your reply means I should use UserStringNop in this case. I do that change, even if i do not understand why its needed.

Reply:
It's there so automated analysis can easily find stringtable keys in code. If a string literal in code will ever be looked up explicitly as a stringtable key, then it needs to appear in a `UserString` or a `UserStringNoOp` wrapper (but not both).